core/shared/src/main/scala/org/http4s/headers/Upgrade-Insecure-Requests.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package headers
+
+import cats.parse.Parser
+import cats.parse.Parser.char
+import org.typelevel.ci._
+
+sealed trait `Upgrade-Insecure-Requests`
```

Comment:
```suggestion
sealed abstract class `Upgrade-Insecure-Requests`
```

Reply:
Honestly it's my personal preference, with this reasoning:

1. I don't see a practical reason it needs to be a `trait` (this is true for most ADTs in my experience)
2. `abstract class`es have theoretically better performance than `trait`s. in any case, it's a cheaper encoding
3. in Scala 3 `enum` is encoded as an `abstract class`, presumably for similar reasons. it also means we could theoretically rewrite as `enum` without breaking binary-compatibility